packages/features/pbac/services/permission-check.service.ts

```diff
@@ -53,8 +58,28 @@ export class PermissionCheckService {
         return [];
       }
 
-      const actions = await this.repository.getResourcePermissions(userId, teamId, resource);
-      return actions.map((action) => PermissionMapper.toPermissionString({ resource, action }));
+      const { membership, orgMembership } = await this.getMembership({ userId, teamId });
+      const actions = new Set<CrudAction | CustomAction>();
+
+      // Get team-level permissions
+      if (membership?.customRoleId) {
+        const teamActions = await this.repository.getResourcePermissionsByRoleId(
```

Comment:
what if we turn `getResourcePermissionsByRoleId` to accept `roleIds`?

Reply:
o.o

Reply:
i think passing in a teamId + an optionalRoleId might be a good shout here. Let me play around with that